src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -1183,37 +1297,84 @@ TEST_F(PolicyHandlerTest, OnGetListOfPermissions) {
   // Arrange
   EnablePolicyAndPolicyManagerMock();
 
-  const uint32_t kAppId_ = 10u;
+  const uint32_t kAppId1_ = 10u;
   const uint32_t kCorId = 1u;
   const std::string default_mac = "00:00:00:00:00:00";
   test_app.insert(mock_app_);
 
   // Expectations
   EXPECT_CALL(*mock_policy_manager_, GetUserConsentForApp(default_mac, _, _));
 
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+  EXPECT_CALL(mock_message_helper_,
               SendGetListOfPermissionsResponse(_, kCorId, _));
 
-  EXPECT_CALL(app_manager_, application(kAppId_))
+  EXPECT_CALL(app_manager_, application(kAppId1_))
       .WillRepeatedly(Return(mock_app_));
 
-  policy_handler_.OnGetListOfPermissions(kAppId_, kCorId);
+  policy_handler_.OnGetListOfPermissions(kAppId1_, kCorId);
 }
 
 TEST_F(PolicyHandlerTest, OnGetListOfPermissions_WithoutConnectionKey) {
   // Arrange
   EnablePolicyAndPolicyManagerMock();
 
-  const uint32_t kAppId_ = 0u;
+  const uint32_t kAppId1_ = 0u;
   const uint32_t kCorId = 1u;
   test_app.insert(mock_app_);
 
   // Expectations
   EXPECT_CALL(app_manager_, applications()).WillRepeatedly(Return(app_set));
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+  EXPECT_CALL(mock_message_helper_,
               SendGetListOfPermissionsResponse(_, kCorId, _));
 
-  policy_handler_.OnGetListOfPermissions(kAppId_, kCorId);
+  policy_handler_.OnGetListOfPermissions(kAppId1_, kCorId);
+}
+
+ACTION_P(SetGroupPermissions, permissions) {
+  arg2 = permissions;
+}
+
+TEST_F(PolicyHandlerTest, OnGetListOfPermissions_GroupPermissions_SUCCESS) {
+  EnablePolicyAndPolicyManagerMock();
+
+  test_app.insert(mock_app_);
+
+  policy::FunctionalGroupPermission group_permission_disallowed1;
+  group_permission_disallowed1.state = GroupConsent::kGroupDisallowed;
+  group_permission_disallowed1.group_alias = "disallowed";
+  group_permission_disallowed1.group_name = "name_disallowed";
+
+  policy::FunctionalGroupPermission group_permission_disallowed2;
+  group_permission_disallowed2.state = GroupConsent::kGroupDisallowed;
+  group_permission_disallowed2.group_alias = "disallowed";
+  group_permission_disallowed2.group_name = "name_disallowed";
+
+  policy::FunctionalGroupPermission group_permission_allowed1;
+  group_permission_allowed1.state = GroupConsent::kGroupAllowed;
+  group_permission_allowed1.group_alias = "allowed";
+  group_permission_allowed1.group_name = "name_allowed";
+
+  policy::FunctionalGroupPermission group_permission_allowed2;
+  group_permission_allowed2.state = GroupConsent::kGroupAllowed;
+  group_permission_allowed2.group_alias = "allowed";
+  group_permission_allowed2.group_name = "name_allowed";
+
```

Comment:
@okozlovluxI think it would be better if you create method for filling this structures